cmd/pilot-agent/main.go

```diff
@@ -99,12 +83,46 @@ var (
 
 			glog.V(2).Infof("proxy role: %#v", role)
 
-			envoyProxy := envoy.MakeProxy(mesh, servicecluster, role.ServiceNode(), configpath, binarypath)
-			agent := proxy.NewAgent(envoyProxy, proxy.DefaultRetry)
-			watcher, err := envoy.NewWatcher(mesh, agent, role)
-			if err != nil {
+			proxyConfig := proxy.DefaultProxyConfig()
+			if config != "" {
+				if err := model.ApplyYAML(config, &proxyConfig); err != nil {
```

Comment:
this is not a convenient thing to launch the agent with an entire YAML as a string. Can you also add a flag to read from a file? We don't have to use it but it will become useful in non pod scenarios, where I just want to run the agent with a bunch of settings.

Reply:
renamed to `configInline` and adding a flag for `configFile` to read from a file.

Reply:
Thanks!